postprocessing/science/compute_isochrones.py

```diff
@@ -0,0 +1,242 @@
+dtSurface = surfaceXdmf.ReadTimeStep()
+
+if not args.faultXdmf:
+    args.faultXdmf = args.filename[:-12]+"fault.xdmf"
+    
+faultXdmf = seissolxdmf.seissolxdmf(args.faultXdmf) 
+faultGeom = faultXdmf.ReadGeometry()
+faultConnect = faultXdmf.ReadConnect()
+faultMidPoints = ComputeTriangleMidpoints(faultGeom, faultConnect)
+dtFault = faultXdmf.ReadTimeStep()
+indexBoundsFault = np.array(args.timeStepRange)
+indexBoundsFault[1] = faultXdmf.ReadNdt()-1 if indexBoundsFault[1] == -1 else indexBoundsFault[1]
+
+# set rupture times before indexBoundsFault[0] to 0:
+ruptureTimes = faultXdmf.ReadData("RT", indexBoundsFault[1]) - faultXdmf.ReadData("RT", indexBoundsFault[0])
+# shift rupture times so that indexBoundsFault[0] corresponds to 0:
+ruptureTimes = np.where(ruptureTimes > 0., ruptureTimes - dtFault * indexBoundsFault[0], 0.)
+
+
+if args.slipRateParameters == "absolute":
+    sliprateParameters = ["ASR"]
+elif args.slipRateParameters == "components":
+    sliprateParameters = ["SRs", "SRd"]
+else:
+    sliprateParameters = ["SRs", "SRd", "ASR"] 
+    
+sliprates = LoadSliprates(sliprateParameters)
+sliprates = np.abs(sliprates)
+
+print(f"surfaceCellBarycenters.shape: {surfaceCellBarycenters.shape}")
+print(f"sliprates.shape: {sliprates.shape}")
+
+stop1 = timeit.default_timer()
+print(f"Time to load data: {stop1 - start:.2f}")
+
+print("Computing travel times and peak slip rate times...")
+
+receiverIndex = ClosestReceiverIndex(np.array([args.x[0], args.y[0], args.z[0]]), surfaceCellBarycenters)
+receiverCoords = surfaceCellBarycenters[receiverIndex]
+
+print(f"Closest receiver: {np.int_(receiverCoords)}")
+
+surfaceVariables = ['u', 'v', 'w'] if 'u' in surfaceXdmf.ReadAvailableDataFields() else ['v1', 'v2', 'v3']
+
+receiverPwaveTraveltime = PTraveltime(receiverIndex, trigger = 0.01)
+print(f"P-wave traveltime to receiver: {receiverPwaveTraveltime:.2f}")
+
+absoluteSliprates = sliprates[-1,:,:] if "ASR" in sliprateParameters else np.linalg.norm(sliprates, axis=0)
+hypocenter = ApproximateHypocenter(ruptureTimes, faultMidPoints)
+print(f"Hypocenter approximated at: {np.round(hypocenter)}")
+
+avgSwaveVelocity = np.linalg.norm(receiverCoords - hypocenter) / receiverPwaveTraveltime / np.sqrt(3) # Assumes a Poisson solid
+print(f"Average S-wave velocity between hypocenter and receiver: {avgSwaveVelocity:.2f}")
+
+allFaultTraveltimes = GetTraveltimesToAllFaultCells(receiverCoords, faultMidPoints, avgSwaveVelocity)
+print(f"Mean traveltime between receiver and fault cells: {np.mean(allFaultTraveltimes):.2f}")
+
+print(f"Sliprate threshold: {args.slipRateThreshold[0]}")
+sliprates = np.where(sliprates >= args.slipRateThreshold[0], sliprates, 0.)  # Set slip rates below the threshold to zero
+peakSliprateTimes = np.argmax(sliprates, axis=2) * dtFault
+
+stop2 = timeit.default_timer()
+print(f"Time to compute values: {stop2 - stop1:.2f}")
+
+if args.medianFilterRadius[0]:
```

Comment:
does this really works?
I would have this only `if args.medianFilterRadius:` works.

Reply:
Yes, it works because it's a required parameter

Reply:
then it is always true.

Reply:
No, you deactivate it by setting it to zero